### ISSUE TITLE
Clean is called twice in Django 1.9

### ISSUE THREAD

**Issue by MadSpindel** [ubernostrum/django-registration#25]
When I upgrade from Django 1.8.6 to Django 1.9 the clean methods in registration/forms.py is called twice (I checked with pdb breakpoint).

I don't know why this is happening (and why it only occurs in Django 1.9) but I believe it's a bug.

**Comment by ubernostrum** [comment]
Are you using any sort of custom `clean` methods? Or just the built-in ones?

**Comment by MadSpindel** [comment]
I believe (I will try tomorrow) if you use eg. RegistrationFormNoFreeEmail with model_activation backend, and put `import pdb; pdb.set_trace()` in the `def clean_email(self):` it will be called twice.

I have som custom code so I could have messed something up so I will try with clean Django 1.9 + django-registration tomorrow, but I don't have this problem in Django 1.8.6 with my current code.

**Comment by MadSpindel** [comment]
Tried with Django 1.9 and a new Django project with master version of django-registration and the clean methods get callled twice. So something is causing the form the clean twice when you are using Django 1.9.

**Comment by MadSpindel** [comment]
Solution:
https://github.com/django/django/commit/e171a83b1516a3a2a393a5300527af1aab21c213

clean was called twice when form was invalid.

So overriding form_invalid solves the problem:
```
def form_invalid(self, form):
        """
        If the form is invalid, re-render the context data with the
        data-filled form and errors.
        """
        return self.render_to_response(self.get_context_data(form=form))
```

**Comment by ubernostrum** [comment]
I'm trying to find out why that wasn't backported into the Django 1.9 branch.

**Comment by ubernostrum** [comment]
In the meantime, 87f25b6 fixes this in django-registration, and will be included in the next release regardless of what we do upstream (since even if Django 1.9.1 fixes this, there'll still be people using 1.9).
